lib/VerticalDelegate.js

```diff
@@ -433,10 +434,10 @@ module.exports = {
 
 		// if we need to refresh, do it now and ensure that we're properly setup to scroll
 		// if we were adding to a partially filled page
-		if (props.index <= end ) this.refresh(list);
+		if (idx <= end ) this.refresh(list);
 		else {
 			// we should confirm that the page which new models are added is need to update list.metrics
-			var lastPageIndex = this.pageForIndex(list, props.index),
+				var lastPageIndex = Object.getOwnPropertyNames(list.metrics.pages).length == 2 ?   this.pageForIndex(list, end) : this.pageForIndex(list, idx),
```

Comment:
What we really need is `accurate page metric` of not generated page.
You assume that this issue could be reproducible only we generated 2 pages.
But If there are some not generate page, we can meet this issue again.